### ISSUE TITLE
use range with time causes an error and crash

### ISSUE THREAD

**Issue by orelHAnyvision** [Hacker0x01/react-datepicker#3673]
**Describe the bug**
dates is not iterable error

**To Reproduce**
use this code snippet:

`const NewDatePicker = ({ variant }) => {
  const [startDate, setStartDate] = useState(new Date())
  const [endDate, setEndDate] = useState(null)
  const onChange = dates => {
    const [start, end] = dates
    setStartDate(start)
    setEndDate(end)
  }

  const ExampleCustomInput = forwardRef(({ value, onClick }, ref) => (
    <button className="example-custom-input" onClick={onClick} ref={ref}>
      {value}
    </button>
  ))

  return (
    <div style={{ height: 600, width: '100%' }}>
      <DatePicker
        selected={startDate}
        onChange={onChange}
        startDate={startDate}
        endDate={endDate}
        selectsRange
        calendarClassName={'calendar-container'}
        className={'date-picker-container'}
        popperClassName={'popper-container'}
        monthsShown={3}
        customInput={<ExampleCustomInput />}
        dateFormat="mm/dd/yyyy hh:mm"
        isClearable
        showTimeInput        
      />
    </div>
  )`

**Expected behavior**
have selection of times for both start and end dates

**Screenshots**

**Desktop**
- OS: mac
- Browser: chrome

**Comment by standbyoneself** [comment]
Same. Any updates on this?

**Comment by phazei** [comment]
Yup, the update is now it completely crashes the calendar, as in it disappears and displays an error message instead if you try to use range and time picker together.

Reproduce:

https://reactdatepicker.com/#example-input-time

add `selectsRange` to the demo code, click the date box and select a date:

```
Error: Minified React error #185; visit https://react.dev/errors/185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
```

**Comment by balajis-qb** [comment]
I'll work on this task

**Comment by martijnrusschen** [comment]
Fix proposed in https://github.com/Hacker0x01/react-datepicker/pull/6079
